src/index.ts

```diff
@@ -142,30 +120,42 @@ export default <ExportedHandler<Env>>{
 											},
 										});
 									}
+									if (!qMessage.ok) {
+										return respondToInteraction({
+											type: InteractionResponseType.ChannelMessageWithSource,
+											data: {
+												content: `Failed to send in question channel, received: ${qMessage.status} ${qMessage.statusText}`,
+												flags: 64,
+											},
+										});
+									}
+									await setDOState(env.ChannelDO, (await qMessage.json<APIMessage>()).id);
 									state.announcement = {
 										channelId: announcement,
 										messageId: (await message.json<APIMessage>()).id,
 									};
 								} else {
 									if (state.announcement) {
-										await fetch(
-											`${RouteBases.api}${Routes.channelMessages(state.announcement.channelId)}/${
-												state.announcement.messageId
-											}`,
-											{
-												headers: {
-													Authorization: `Bot ${env.DISCORD_BOT_TOKEN}`,
-													"Content-Type": "application/json",
-												},
-												method: "DELETE",
-											}
+										await deleteMessage(
+											state.announcement.channelId,
+											state.announcement.messageId,
+											env.DISCORD_BOT_TOKEN
 										);
 									}
 									state.announcement = undefined;
+									const questionId = await getDOState(env.ChannelDO);
+									if (questionId) {
+										await deleteMessage(
+											env.DISCORD_QUESTION_CHANNEL,
+											questionId,
+											env.DISCORD_BOT_TOKEN
+										);
+									}
+									ctx.waitUntil(deleteDOState(env.ChannelDO));
```

Comment:
We can move this into the `if (questionId)` block

Reply:
Derp(done).